src/v/cloud_storage/topic_manifest.cc

```diff
@@ -22,12 +23,175 @@
 
 #include <boost/lexical_cast.hpp>
 #include <fmt/ostream.h>
+#include <rapidjson/encodings.h>
+#include <rapidjson/error/en.h>
 #include <rapidjson/istreamwrapper.h>
 #include <rapidjson/ostreamwrapper.h>
+#include <rapidjson/reader.h>
 
 #include <chrono>
+#include <stdexcept>
+#include <string_view>
 
 namespace cloud_storage {
+
+struct topic_manifest_handler
+  : public rapidjson::
+      BaseReaderHandler<rapidjson::UTF8<>, topic_manifest_handler> {
+    topic_manifest_handler()
+      : _state(expectManifestStart) {}
+
+    bool StartObject() {
+        vlog(cst_log.trace, "StartObject(), state_: {}", _state);
+        switch (_state) {
+        case expectManifestStart:
+            _state = expectKey;
+            return true;
+        default:
+            return false;
+        }
+    }
+
+    bool Key(const char* str, rapidjson::SizeType length, bool copy) {
+        vlog(
+          cst_log.trace,
+          "Key({}, {}, {}), state_: {}",
+          str,
+          length,
+          copy,
+          _state);
+        switch (_state) {
+        case expectKey:
+        case expectKeyOrObjectEnd:
+            _key = std::string(str, length);
+            _state = expectValue;
+            return true;
+        default:
+            return false;
+        }
+    }
+
+    bool String(const char* str, rapidjson::SizeType length, bool copy) {
+        vlog(
+          cst_log.trace,
+          "String({}, {}, {}), state_: {}",
+          str,
+          length,
+          copy,
+          _state);
+        switch (_state) {
+        case expectValue:
+            if ("namespace" == _key) {
+                _topic_config.tp_ns.ns = model::ns(std::string(str, length));
```

Comment:
can we get rid of constructing std::string here (e.g. by using std::string_view)?

Reply:
yes, changed to create `std::string_view` early